### PR TITLE
Add check on master of running and total

### PR DESCRIPTION
**Pull request by BlueShells**
add total and running pod check

### CODE REVIEW

**Comment by BlueShells** [comment]
@twiest ,could you review this for me ,I done the local check already,zabbix get the value

File: scripts/monitoring/cron-send-user-pod-count.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      echo "Running: ops-zagg-client -k 'openshift.master.pod.total.count' -o '$COUNT_TOTAL'"
+      ops-zagg-client -k "openshift.master.pod.total.count" -o "$COUNT_TOTAL"
+      echo 
+  else
+      echo "Only RUNNING or TOTAL is acceptable"
+  fi
+
+else
+  echo "Usage:`basename $0` RUNNING|TOTAL"
+fi
```
Comment (twiest) [inline scripts/monitoring/cron-send-user-pod-count.sh]: Why not just always send both running and total pod counts to zabbix?

It's fine to send multiple values in the same script. In fact, at some point, we're going to combine a bunch of these together.

File: docker/oso-rhel7-zagg-client/root/config.yml [hunk 2]
```diff
@@ -67,6 +67,15 @@
     - name: send openshift-master pods count
       hour: "*/2"
       job: ops-runner -n cspodc.openshift.master.pod.count /usr/bin/cron-send-pod-count
+    - name: send cluster user running pod total count
+      hour: "*/2"
+      job: ops-runner -n csurpc.openshift.master.user.pod.running.count /usr/bin/cron-send-user-pod-count "RUNNING"
+    - name: send cluster user pod total count
+      hour: "*/2"
+      job: ops-runner -n csurpc.openshift.master.user.pod.total.count /usr/bin/cron-send-user-pod-count "TOTAL"
```
Comment (twiest) [inline docker/oso-rhel7-zagg-client/root/config.yml]: Here there are 2 separate cron jobs. I think there should just be 1.

Something like this:
```
    - name: send cluster user pod total count
      hour: "*/2"
      job: ops-runner -n csurpc.openshift.master.user.pod.count /usr/bin/cron-send-user-pod-count
```

File: scripts/monitoring/cron-send-user-pod-count.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash -e
+
+if [ $# -eq 1 ]; then
+  if [[ "RUNNING" = $1 ]]; then
```
Comment (twiest) [inline scripts/monitoring/cron-send-user-pod-count.sh]: Don't do either of these if statements. Don't look at $1 at all.

File: scripts/monitoring/cron-send-user-pod-count.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      COUNT_RUNNING=$(KUBECONFIG=/etc/openshift/master/admin.kubeconfig oadm manage-node --list-pods --selector=type=compute | grep Running | wc -l  ; exit ${PIPESTATUS[0]})
+      echo
+      echo "Number of User Running pod account : $COUNT_RUNNING"
+      echo
+      echo "Running: ops-zagg-client -k 'openshift.master.user.pod.running.count' -o '$COUNT_RUNNING'"
+      ops-zagg-client -k "openshift.master.user.pod.running.count" -o "$COUNT_RUNNING"
+      echo
+  elif [[ "TOTAL" = $1 ]]; then
```
Comment (twiest) [inline scripts/monitoring/cron-send-user-pod-count.sh]: Don't do this if statement. Don't look at $1 at all.

File: scripts/monitoring/cron-send-user-pod-count.sh (hunk 1 above)
Comment (twiest) [inline scripts/monitoring/cron-send-user-pod-count.sh]: Remove these "else and fi" to line up with the removal of the if statements above.

File: scripts/monitoring/cron-send-user-pod-count.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+      echo
+      echo "Running: ops-zagg-client -k 'openshift.master.user.pod.running.count' -o '$COUNT_RUNNING'"
+      ops-zagg-client -k "openshift.master.user.pod.running.count" -o "$COUNT_RUNNING"
+      COUNT_TOTAL=$(KUBECONFIG=/etc/openshift/master/admin.kubeconfig oadm manage-node --list-pods --selector=type=compute | wc -l  ; exit ${PIPESTATUS[0]})
+      echo
+      echo "Number of User pod Total account : $COUNT_TOTAL"
+      echo
+      echo "Running: ops-zagg-client -k 'openshift.master.pod.total.count' -o '$COUNT_TOTAL'"
+      ops-zagg-client -k "openshift.master.pod.total.count" -o "$COUNT_TOTAL"
+      echo
```
Comment (twiest) [inline scripts/monitoring/cron-send-user-pod-count.sh]: You need to remove the indentation since the if statements are now gone.

**Comment by twiest** [comment]
@BlueShells Looks great. I added one more minor cleanup comment.

**Comment by twiest** [comment]
:+1:

**Comment by mwoodson** [comment]
Please do not merge this branch.  The checks here are now handled inside of a new master check that has combined multiple master checks, including this one.

The new script can be found in this pr:

https://github.com/openshift/openshift-tools/pull/190

Thanks!

Matt

**Pull request closed without merging**